specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-05-01/frontdoor.json

```diff
@@ -0,0 +1,2348 @@
+          "$ref": "./network.json#/definitions/SubResource"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/RouteConfiguration"
+        }
+      ]
+    },
+    "RedirectConfiguration": {
+      "description": "Describes Redirect Route.",
+      "x-ms-discriminator-value": "#Microsoft.Azure.FrontDoor.Models.FrontdoorRedirectConfiguration",
+      "properties": {
+        "redirectType": {
+          "description": "The redirect type the rule will use when redirecting traffic.",
+          "type": "string",
+          "enum": [
+            "Moved",
+            "Found",
+            "TemporaryRedirect",
+            "PermanentRedirect"
+          ],
+          "x-ms-enum": {
+            "name": "FrontDoorRedirectType",
+            "modelAsString": true
+          }
+        },
+        "redirectProtocol": {
+          "description": "The protocol of the destination to where the traffic is redirected",
+          "type": "string",
+          "enum": [
+            "HttpOnly",
+            "HttpsOnly",
+            "MatchRequest"
+          ],
+          "x-ms-enum": {
+            "name": "FrontDoorRedirectProtocol",
+            "modelAsString": true
+          }
+        },
+        "customHost": {
+          "type": "string",
+          "description": "Host to redirect. Leave empty to use the incoming host as the destination host."
+        },
+        "customPath": {
+          "type": "string",
+          "description": "The full path to redirect. Path cannot be empty and must start with /. Leave empty to use the incoming path as destination path."
+        },
+        "customFragment": {
+          "type": "string",
+          "description": "Fragment to add to the redirect URL. Fragment is the part of the URL that comes after #. Do not include the #."
+        },
+        "customQueryString": {
+          "description": "The set of query strings to be placed in the redirect URL. Setting this value would replace any existing query string; leave empty to preserve the incoming query string. Query string must be in <key>=<value> format. The first ? and & will be added automatically so do not include them in the front, but do separate multiple query strings with &.",
+          "type": "string"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/RouteConfiguration"
+        }
+      ]
+    },
+    "Backend": {
+      "description": "Backend address of a frontDoor load balancer.",
+      "type": "object",
+      "properties": {
+        "address": {
+          "description": "Location of the backend (IP address or FQDN)",
+          "type": "string"
+        },
+        "privateLinkAlias": {
+          "description": "The Alias of the Private Link resource. Populating this optional field indicates that this backend is 'Private'",
+          "type": "string"
+        },
+        "privateLinkResourceId": {
```

Comment:
>privateLinkResourceId [](start = 9, length = 21)

can this and privateLinkAlias be populated at the same time?

Reply:
They could be. However, RP would then let the user know to use either one